src/metabase/models/table.clj

```diff
@@ -127,6 +131,73 @@
   ([_ pk]
    (mi/can-write? (t2/select-one :model/Table pk))))
 
+;;; ------------------------------------------------ SQL Permissions ------------------------------------------------
+
+(def ^:private TableVisibilityConfig
+  [:map
+   [:user-id pos-int?]
+   [:is-superuser? :boolean]
+   [:view-data-permission-level data-perms/PermissionValue]
+   [:create-queries-permission-level data-perms/PermissionValue]])
+
+(mu/defn- perm-type-to-int-case
+  [perm-type :- data-perms/PermissionType column-or-perm-value :- [:or :keyword [:tuple [:= ::h2x/literal] :string]]]
```

Comment:
nit - might be worth pulling `[:tuple [:= ::h2x/literal] :string]` out into a dedicated schema in `metabase.util.honey-sql-2`?

Reply:
I also find malli-annotated arg lists to be more readable with each arg on a separate line, but up to you